api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -486,4 +486,10 @@ message CertificateValidationContext {
   // If this option is set to true, only the certificate at the end of the
   // certificate chain will be subject to validation by :ref:`CRL <envoy_v3_api_field_extensions.transport_sockets.tls.v3.CertificateValidationContext.crl>`.
   bool only_verify_leaf_cert_crl = 14;
+
+  // Config for the max number of intermediate certificates in chain that are parsed during verification
+  // This does not include the leaf certificate. If configured, and the certificate chain is longer than allowed, the certificates
+  // above the limit are ignored, and certificate validation will fail.
+  // https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_verify_depth.html
+  uint32 max_verify_depth = 16 [(validate.rules).uint32 = {lte: 100}];
```

Comment:
The reason is to prevent a client from consuming excess CPU by presenting a client cert with an intentionally long chain. If the server knows all possible intermediate chains and thus knows the maximum possible chain length, this can be set to prevent bothering to try validating longer chains.

It might make more sense to use a `google.protobuf.UInt32Value` because most users will probably accept the default. I'm not sure about the tradeoffs of 0==default vs unset==default; that's a question for api-shephards.

Reply:
What would the default be? I agree with using a WKT and setting some reasonable default? Note that the use of a non-zero default itself will have to be temporarily feature flagged but that's probably fine.

Reply:
The default is in BoringSSL, and we use it by not calling the relevant API at all. So this PR, if the new setting isn't set, will result in no behavior change at all, so shouldn't need a feature flag.

Reply:
OK makes sense. Yeah I would use a WKT to be mean no change to the default.

Reply:
would it be simpler/more intuitive to rename this to set_max_verify_depth? So no config -> no setting and keep the default?

Reply:
I think the existing name is more consistent with our other APIs so I would recommend not changing it.

Reply:
ok. so  just add a SSL_MAX_DEPTH_ DEFAULT indicating no change; will add.

Reply:
I would still make it a WKT (well known type) instead of a uint32, just for clarity that if unset you get the default.

Reply:
Ack.. changing to well-known-type